src/krotov/propagators.py

```diff
@@ -251,13 +257,21 @@ def __call__(
         self._t += dt
         self._r.integrate(self._t)
         self._y = self._r.y
-        return qutip.Qobj(
-            dense2D_to_fastcsr_fmode(
-                vec2mat(self._y), state.shape[0], state.shape[1]
-            ),
-            dims=state.dims,
-            isherm=True,
-        )
+        if is_qutip5:
+            return qutip.Qobj(
+                unstack_columns(self._y),
+                dims=state.dims,
+                isherm=True,
+                dtype="csr"
+            )
```

Comment:
`dtype` is not a parameter of `Qobj`.
Is that output really sparse? 
If the output need to be converted, use `to`:
```suggestion
            return qutip.Qobj(
                unstack_columns(self._y),
                dims=state.dims,
                isherm=True,
            ).to("CSR")
```

Reply:
Just as a note (and this could be a separate PR): The sparse storage of QuTiP objects was probably a major source of slowness in krotov. Most usages of this package would be for small Hilbert spaces – certainly all the examples are for small Hilbert spaces. If I followed QuTiP's development correctly, version 5 should have the ability to use dense storage internally, and it would be good to use that as much as possible here (in the examples). Of course, if people manually define their objectives with sparse objects, those should still be respected.

Reply:
Quite possibly… the notebooks definitely serve as "integration tests". With my update of the tooling, you should be able to run it locally and/or the CI should run it for you.